### PR TITLE
Use pretty 404 pages in repo.HTTPBackend

### PR DESCRIPTION
**Pull request by fnkr**
**Before**:

**After**:

### CODE REVIEW

File: routers/repo/http.go [hunk 1]
```diff
@@ -433,8 +433,6 @@ func sendFile(contentType string, hr handler) {
 	w, r := hr.w, hr.r
 	reqFile := path.Join(hr.Dir, hr.File)
 
-	// fmt.Println("sendFile:", reqFile)
```
Comment (fnkr) [inline routers/repo/http.go]: Removed this comment too.

**Comment by Unknwon** [comment]
Maybe `renderNotFound` could be removed too?

**Comment by fnkr** [comment]
@Unknwon Actually, I tried to archive that but it's still used by `sendFile()` which is used by 6 functions that are used by [this struct](https://github.com/gogits/gogs/blob/v0.8.25/routers/repo/http.go#L251-L263) that [is used to add routes to the URL router](https://github.com/gogits/gogs/blob/v0.8.25/routers/repo/http.go#L291), that are used by the Git client (I believe), and I didn't wanted to return rendered HTML to Git.

**Comment by Unknwon** [comment]
I'm confused by your comments and your changes?

>  that is used to add routes to the URL router, that are used by the Git client (I believe), and I didn't wanted to return rendered HTML to Git as Git probably prints it to the console.

Didn't you just changes those?

**Comment by fnkr** [comment]
[`renderNotFound`](https://github.com/gogits/gogs/blob/v0.8.25/routers/repo/http.go#L527-L530) was used two functions: `HTTPBackend()` and `sendFile()`. This PR just changes `HTTPBackend()` to use `ctx.Handle()` instead of `renderNotFound()`. `sendFile()` still uses `renderNotFound()`.

**Comment by Unknwon** [comment]
OK, cool!

**Pull request merged**